components/logins/src/db.rs

```diff
@@ -25,6 +28,10 @@ use sync15::{
 
 pub struct LoginDb {
     pub db: Connection,
+    // The logins DB has exactly 1 interrupt scope, but we keep a copy of the
+    // Arc with the counter so we can create an interrupt handle.
+    interrupt_counter: Arc<AtomicUsize>,
+    pub interrupt_scope: SqlInterruptScope,
```

Comment:
Yeah I don't think the scope should live here.

Instead we should add a new `struct HistoryStore<'a> { &'a LoginsDb, SqlInterruptScope }`